pkg/governance-scripts/contracts/BalancerContractRegistryInitializer.sol

```diff
@@ -86,11 +96,25 @@ contract BalancerContractRegistryInitializer {
             IBalancerContractRegistry.addOrUpdateBalancerContractAlias.selector
         );
 
-        _authorizer.grantRole(registerContractRole, address(this));
-        _authorizer.grantRole(addAliasRole, address(this));
+        // Ensure the contract has been granted the required permissions, given the deployment parameters.
+        uint256 numPoolFactories = _poolFactoryNames.length;
+        uint256 numRouters = _routerNames.length;
+        uint256 numAliases = _aliasNames.length;
+
+        if (
+            ((numRouters > 0 || numPoolFactories > 0) &&
+                _authorizer.canPerform(registerContractRole, address(this), address(balancerContractRegistry)) ==
+                false) ||
+            (numAliases > 0 &&
+                _authorizer.canPerform(addAliasRole, address(this), address(balancerContractRegistry)) == false)
```

Comment:
Do you think it's worth adding a getter that returns `true` if both permissions are granted and `false` otherwise?

you can always check directly on the authorizer though, maybe too much.

Reply:
It's possible they don't need to grant both permissions. Maybe they don't want to use aliases (or not on a certain chain). Or maybe they *only* need aliases, because they already manually registered the contracts before this was deployed.